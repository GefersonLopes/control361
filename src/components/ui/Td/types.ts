export interface TdProps extends React.HTMLAttributes<HTMLTableCellElement> {
  children: React.ReactNode;
  className?: string;
}
