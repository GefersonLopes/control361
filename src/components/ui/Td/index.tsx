import clsx from "clsx";

import type { TdProps } from "./types";

function Td({ children, className }: TdProps) {
  return (
    <td className={clsx("px-4 py-2 whitespace-nowrap", className)}>
      {children}
    </td>
  );
}

export default Td;
