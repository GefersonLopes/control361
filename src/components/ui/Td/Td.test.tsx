import "@testing-library/jest-dom";

import { render, screen } from "@testing-library/react";

import Td from "../Td";

describe("Td component", () => {
  it("renders a td element with default classes and children", () => {
    render(
      <table>
        <tbody>
          <tr>
            <Td>Cell Content</Td>
          </tr>
        </tbody>
      </table>,
    );
    const cell = screen.getByText("Cell Content");
    expect(cell.tagName).toBe("TD");
    expect(cell).toHaveClass(
      "px-4",
      "h-[40px]",
      "whitespace-nowrap",
      "text-center",
      "text-sm",
      "text-white",
      "text-normal",
      "tracking-wide",
      "font-inter",
    );
  });

  it("merges a custom className when provided", () => {
    render(
      <table>
        <tbody>
          <tr>
            <Td className="extra-class another-class">Custom Cell</Td>
          </tr>
        </tbody>
      </table>,
    );
    const cell = screen.getByText("Custom Cell");
    expect(cell).toHaveClass("extra-class", "another-class");
  });

  it("passes arbitrary HTML props to the td element", () => {
    render(
      <table>
        <tbody>
          <tr>
            <Td data-testid="td-test" aria-label="label" title="my-title">
              Test
            </Td>
          </tr>
        </tbody>
      </table>,
    );
    const cell = screen.getByTestId("td-test");
    expect(cell).toHaveAttribute("aria-label", "label");
    expect(cell).toHaveAttribute("title", "my-title");
  });

  it("supports the colSpan prop", () => {
    render(
      <table>
        <tbody>
          <tr>
            <Td colSpan={3}>Spanned Cell</Td>
          </tr>
        </tbody>
      </table>,
    );
    const cell = screen.getByText("Spanned Cell");
    expect(cell).toHaveAttribute("colspan", "3");
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <table>
        <tbody>
          <tr>
            <Td className="snapshot-class" colSpan={2} data-extra="x">
              Snap
            </Td>
          </tr>
        </tbody>
      </table>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
