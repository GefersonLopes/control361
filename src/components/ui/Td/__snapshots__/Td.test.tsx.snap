// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Td component matches the snapshot 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td
          class="px-4 h-[40px] whitespace-nowrap text-center text-sm text-white text-normal tracking-wide font-inter snapshot-class"
          colspan="2"
          data-extra="x"
        >
          Snap
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
